### PR TITLE
Lazy bootstrap cleanup & basic improvements

### PR DESCRIPTION
**Pull request by SergiySW**
~Based on https://github.com/nanocurrency/nano-node/pull/2290~
Also contains blocks count cache quick fix

### CODE REVIEW

File: nano/node/bootstrap/bootstrap.cpp [hunk 1]
```diff
@@ -311,7 +307,8 @@ unsigned nano::bootstrap_attempt::target_connections (size_t pulls_remaining)
 
 	// Only scale up to bootstrap_connections_max for large pulls.
 	double step = std::min (1.0, std::max (0.0, (double)pulls_remaining / nano::bootstrap_limits::bootstrap_connection_scale_target_blocks));
-	double target = (double)node->config.bootstrap_connections + (double)(node->config.bootstrap_connections_max - node->config.bootstrap_connections) * step;
+	double lazy_factor = (mode == nano::bootstrap_mode::lazy) ? (double)node->config.bootstrap_connections : 0.0;
+	double target = (double)node->config.bootstrap_connections + (double)(node->config.bootstrap_connections_max - node->config.bootstrap_connections) * step + lazy_factor;
```
Comment (cryptocode) [inline nano/node/bootstrap/bootstrap.cpp]: nitpick: lazy_factor is not a factor, maybe use `lazy_term` and `step_factor` (or scale) in this expression.

File: nano/node/bootstrap/bootstrap.cpp [hunk 2]
```diff
@@ -732,139 +743,126 @@ bool nano::bootstrap_attempt::process_block_lazy (std::shared_ptr<nano::block> b
...
+		{
+			lazy_add (block_a->source ());
+		}
+		else if (block_a->type () == nano::block_type::state)
+		{
+			lazy_block_state (block_a);
+		}
+		lazy_blocks.insert (hash);
+		// Adding lazy balances for first processed block in pull
+		if (pull_blocks == 0 && (block_a->type () == nano::block_type::send || block_a->type () == nano::block_type::send))
```
Comment (cryptocode) [inline nano/node/bootstrap/bootstrap.cpp]: block_a->type () == nano::block_type::send is repeated twice. Was `state` meant?

Comment (SergiySW) [reply nano/node/bootstrap/bootstrap.cpp]: Sure

File: nano/node/bootstrap/bootstrap.cpp (hunk 1 above)
Comment (SergiySW) [reply nano/node/bootstrap/bootstrap.cpp]: Thanks

**Pull request merged**